### PR TITLE
Introduce backup fetch feature

### PR DESCRIPTION
**Pull request by wiktor2200**
Hi! I've added feature to download created backup from remote to local host. It enables user to turn it on/off and define path where to save downloaded file on local machine.
There is one caveat here, Ansible user has to be the owner of backup directory, because of known `fetch` limitations while using `become`.
I've also added README entry, how to set it properly.
@aalaesar please take a look :)

### CODE REVIEW

**Comment by wiktor2200** [comment]
This PR has to be merged after #1 and #5.

**Review by aalaesar — CHANGES_REQUESTED** [review]
I had to use my time machine to fix the mess done by my gitea mirror. and I also deleted the test folder from History. please rebase (again) your branch please.
I'm not sure but you may need to discard the old commit history from master first
sorry 😞

**Comment by wiktor2200** [comment]
No problem, fixed ;)

**Comment by wiktor2200** [comment]
I fixed this too, this has to be merged after #5

**Comment by aalaesar** [comment]
I merged #5 but its commits still appear in this PR.
I guess you have to rebase :-|

File: tasks/fetching.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
```
Comment (aalaesar) [inline tasks/fetching.yml:1]: I don't see a patch on `tasks/main.yml` to include this file.
Did it got deleted during the numerous rebases ? 😖

**Review by aalaesar — CHANGES_REQUESTED** [review]

File: tasks/fetching.yml (hunk 1 above)
Comment (wiktor2200) [reply tasks/fetching.yml:1]: Exactly, I've fixed it + fast-forwarded to current master :)

File: tasks/fetching.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+- name: Fetch file from remote to local
+  become: false # Using become may cause OOM errors. (https://docs.ansible.com/ansible/latest/collections/ansible/builtin/fetch_module.html#notes)
+  ansible.builtin.fetch:
+    src: "{{ nc_archive_path }}.{{ nextcloud_backup_format }}"
+    dest: "{{ nextcloud_backup_fetch_local_path }}"
+  when: nextcloud_backup_fetch_to_local
```
Comment (aalaesar) [inline tasks/fetching.yml]: this seems to be a duplicate condition now with the same one in main.yml

Comment (wiktor2200) [reply tasks/fetching.yml]: Good catch, fixed :)

Comment (wiktor2200) [reply tasks/fetching.yml]: Good catch, thanks. Fixed :)

**Review by aalaesar — APPROVED** [review]

**Pull request merged**